packages/massa-web3/src/web3/SmartContractsClient.ts

```diff
@@ -405,6 +424,42 @@ export class SmartContractsClient
     )
   }
 
+  public async watchOperationStatus(
+    opId: string,
+    callback: (status: EOperationStatus) => void,
+    timeInterval = TX_POLL_INTERVAL_MS,
+    timeout = Date.now() + WAIT_OPERATION_TIMEOUT
+  ): Promise<() => void> {
+    let lastStatus = await this.getOperationStatus(opId)
+    callback(lastStatus)
+
+    const interval = setInterval(async () => {
+      if (Date.now() > timeout) {
+        clearInterval(interval)
+        return
+      }
+
+      try {
+        const newStatus = await this.getOperationStatus(opId)
+        if (newStatus !== lastStatus) {
+          lastStatus = newStatus
+          callback(newStatus)
+        }
+
+        if (
+          newStatus === EOperationStatus.FINAL_SUCCESS ||
+          newStatus === EOperationStatus.FINAL_ERROR
+        ) {
+          clearInterval(interval)
+        }
+      } catch (error) {
+        console.warn(error)
```

Comment:
linter says `Unexpected console statement` maybe consider removing this rule if we deliberately want to console.log

Reply:
This is just a warning, and I think we should keep it for now, as it prevents us from forgetting debug purpose console.log.

Reply:
Ok I understand you opinion. I didn't added the rule myself so I need to think about it and decide if we really want to use console here

Reply:
I tried another approche that allow the user to have more control what do you think ?